### ISSUE TITLE
ingress-nginx not work good with multi-instances on kubernetes cluster

### ISSUE THREAD

**Issue by snup67** [razidare/Trainings#1]
related to you movie [https://www.youtube.com/watch?v=4CrMF0PqoWY](url) that was very good I want to upload a problem that I faced with ingress-nginx for multi-instances on kubernetes cluster.

got the GUI of Jenkins but if click on management I get to nginx console, since the rewrite is to main application . 
I notes that Jenkins management icon redirect to //management, in my case jenkins-ci/management

I don't get any idea how to solve it.
**I think I can't use ingress-nginx controller because he can't handle rewrite well with Jenkins application**

**The jenkins ingress**
![image](https://user-images.githubusercontent.com/25963382/213576013-67d32dcf-4b86-4259-bd8e-2e34da028699.png)
**ingress describe**
![image](https://user-images.githubusercontent.com/25963382/213576139-6aca1e9a-e311-4af8-94c5-d6cba531279e.png)
**The jenkins console**
![image](https://user-images.githubusercontent.com/25963382/213577141-00bef393-9d61-4529-82ed-6848a666e832.png)

**Comment by razidare** [comment]
Hi, Snup,

I don't think I understand the problem, can you rephrase it like:

Current behavior:
Expected behavior:

But my advice, for the first ingress path, instead of /, also try using
(/|$)(.*)

În vin., 20 ian. 2023 la 00:34, snup67 ***@***.***> a scris:

**Comment by snup67** [comment]
I also used that (/|$)(.*) and $2 on the annotation rewrite - didn't helped.
see the screenshot of Jenkins console GUI (I attached it) - see in the bottom the path if you click manage in the console, in such a case and such configure I'm not getting to the path inside Jenkins

I saw also Skipper controller but didn't use it.

**Comment by razidare** [comment]
Hi, Snup,

I'm more than happy to try and help but I don't understand what you want to achieve.
Can you please rephrase like:

Current behavior:
Expected behavior:

For example:

Expected behavior: When I click on X button, I want to be taken to Y URL
Current behavior: When I click on X button, I am taken to Z URL

Give as much detail as possible so that I can understand your situation better. From what I'm looking at, you get the right behavior from the current configuration.

**Comment by snup67** [comment]
I'm trying to achieve this scenario:

Lets assume that I got thejenkins console when I put **jenkins-ci/test** in chrome  web browser URL address.
Now, I click on **manage** jenkins.
 **I'm not moving to the correct page inside the jenkins application**, 
instead I'm moving to **jenkins-ci** URL (the default ingress page).

clicking on any button not move to the correct page, it moves to jenkins-ci page (the default ingress page)
the rewrite-target annotation is not working in such configuration (see screenshot of ingress describe command)

**Comment by razidare** [comment]
I'll rephrase to see if I got the question right:

When you navigate to **jenkins-ci/test** you reach this page:
![image](https://user-images.githubusercontent.com/36229102/213984978-0c406981-2509-464f-99b6-b01c24a5629f.png)

Then, you want to click on the **Manage Jenkins** button, which should take you to a different management page, but instead,  you go to the default page (the / configuration of the ingress). 
Is that correct?

**Comment by snup67** [comment]
yes, you are correct

**Comment by razidare** [comment]
Ok, so here's what I understand from your configuration:

Your host is **_jenkins-ci_**, so every request you make to this prefix will take one of 3 possible routes:

1. / (matches anything)
2. /jenkins/clean/*
3. /jenkins/test/*

From your screenshot, I see that the console you have opened has the URL **_jenkins-ci/jenkins/clean_** - can you please confirm this?
![image](https://user-images.githubusercontent.com/36229102/214008934-4679debb-476d-4c90-96b3-ee1691ec1c1f.png)

Now, from your screenshot, it looks like when you hover the "Manage Jenkins" button, it shows you the path it will redirect to in the lower left corner - and that path is **_jenkins-ci/manage_**
![image](https://user-images.githubusercontent.com/36229102/214009234-5f1229af-dd94-4be4-ae1f-733e2210bb88.png)
 
 The **_jenkins-ci/manage_** will match rule 1. (/) which should take you here:
![image](https://user-images.githubusercontent.com/36229102/214009652-f12c4256-761d-4962-8afa-edbb268736e9.png)

Are you trying to get to **_jenkins-ci/jenkins/clean/manage_** instead?
Can you please try opening the  **_jenkins-ci/jenkins/clean/manage_** page in your browser? Is this where you want to get when you click the button?

**Comment by snup67** [comment]
Not working
 **jenkins-ci/jenkins/clean/manage** bring me to **jenkins-ci**
 
 any button on jenkins console (Like manage) brings me back to **jenkins-ci** page and not to jenkins page

**Comment by razidare** [comment]
Not sure what you mean by "jenkins" page. All your ingress paths start with the prefix "jenkins-ci":
![image](https://user-images.githubusercontent.com/36229102/214033910-d8705af5-0efa-44f4-be92-8ba049e9e4ec.png)

What would you expect the URL to the "jenkins" page be?

**Comment by snup67** [comment]
When you are in the console of any application you can browse from page to page in the application.
in my case any button on Jenkins console get you to ingress main page instead to the correct page of Jenkins.
